src/zenml/materializers/default_materializer_registry.py

```diff
@@ -71,10 +71,24 @@ def __getitem__(self, key: Type[Any]) -> Type["BaseMaterializer"]:
         if key in self.materializer_types:
             return self.materializer_types[key]
         else:
-            raise KeyError(
-                f"Type {key} does not have a default `Materializer`! Please "
-                f"specify your own `Materializer`."
-            )
+            compatible_subclasses = {
+                self.materializer_types[t]
+                for t in self.materializer_types
+                if issubclass(key, t)
+            }
+            if len(compatible_subclasses) == 1:
+                return self.materializer_types[compatible_subclasses[0]]
+            else:
+                raise KeyError(
+                    f"Type {key} is subclassing more than one type, thus it "
```

Comment:
Maybe we can also include the `compatible_subclasses` in the error somehow to show which materializers overlap?